Try to improve React dependencies handling

zzarcon opened the pull request:
Changes are related to https://github.com/skatejs/skatejs/pull/808. Regarding the **benchmark** library, I just removed from the `files` since we are not using the [bench file](https://github.com/skatejs/skatejs/blob/master/test/lib/bench.js), if we want to add it again we can just put it as a npm dependency.

treshugart commented on webpack.config.js:
```diff
@@ -32,6 +32,11 @@ const standardConfig = {
     }],
   },
   plugins: [
+    new webpack.DefinePlugin({
```
This should probably only be enabled in `--perf` mode.

zzarcon replied:
Good point

treshugart reviewed (changes requested):
Could you also make sure your commits follow conventional changelog please (git cz or sk-commit)? Thanks!

zzarcon committed:
fix(): Make dependencies use production build when running perf test

Define a Webpack plugin to force production environment in order to get production dependencies

zzarcon commented:
@treshugart I applied the changes and committed them using `git cz` properly, let me know if it's fine now
